Runtimes/Core/cmake/modules/EmitSwiftInterface.cmake

```diff
@@ -9,63 +9,74 @@ function(emit_swift_interface target)
   # Generate the target-variant binary swift module when performing zippered
   # build
   if(SwiftCore_VARIANT_MODULE_TRIPLE)
-    set(variant_module_tmp_dir "${CMAKE_CURRENT_BINARY_DIR}/${target}-${SwiftCore_VARIANT_MODULE_TRIPLE}")
-    file(MAKE_DIRECTORY "${variant_module_tmp_dir}")
+    # Clean this up once CMake has nested swiftmodules in the build directory:
+    # https://gitlab.kitware.com/cmake/cmake/-/merge_requests/10664
+
+    # We can't expand the Swift_MODULE_NAME target property in a generator
+    # expression or it will fail saying that the target doesn't exist.
+    get_target_property(module_name ${target} Swift_MODULE_NAME)
+    if(NOT module_name)
+      set(module_name ${target})
+    endif()
     target_compile_options(${target} PRIVATE
-      "$<$<COMPILE_LANGUAGE:Swift>:SHELL:-emit-variant-module-path ${variant_module_tmp_dir}/${target}.swiftmodule>")
+      "$<$<COMPILE_LANGUAGE:Swift>:SHELL:-emit-module-path ${CMAKE_CURRENT_BINARY_DIR}/${module_name}.swiftmodule/${SwiftCore_MODULE_TRIPLE}.swiftmodule>"
+      "$<$<COMPILE_LANGUAGE:Swift>:SHELL:-emit-variant-module-path ${CMAKE_CURRENT_BINARY_DIR}/${module_name}.swiftmodule/${SwiftCore_VARIANT_MODULE_TRIPLE}.swiftmodule>")
+    add_custom_command(OUTPUT "${CMAKE_CURRENT_BINARY_DIR}/${module_name}.swiftmodule/${SwiftCore_MODULE_TRIPLE}.swiftmodule"
+      DEPENDS ${target})
+    target_sources(${target}
+      INTERFACE
+        $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/${module_name}.swiftmodule/${SwiftCore_MODULE_TRIPLE}.swiftmodule>)
   endif()
 
   # Generate textual swift interfaces is library-evolution is enabled
   if(SwiftCore_ENABLE_LIBRARY_EVOLUTION)
```

Comment:
Should we consider removing this? The standard library build does emit the textual interface for all targets and the work that Alexis is doing to always use the textual interface.